iGCS/UserInterface/ViewControllers/GCSMapViewController.m

```diff
@@ -157,6 +165,29 @@ - (BOOL)shouldAutorotateToInterfaceOrientation:(UIInterfaceOrientation)interface
     return YES;
 }
 
+- (void)dealloc {
+    [[NSNotificationCenter defaultCenter] removeObserver:self];
+}
+
+#pragma mark - 
+#pragma mark Handle NSnotifications for UI updates
+
+// These will get called only when there is a change event
+// instead of at every tick of the telemetry data
```

Comment:
I guess this is a reflection of how long you've wanted this, but it doesn't require a comment :-)

As a first time reader, I'd find this comment to be confusing precisely because there is no relationship to telemetry here ("So why am I being warned about telemetry? I'd better look again.")